### PR TITLE
[jog] Autodiscover bare graphs when initializing dagit

### PR DESCRIPTION
**Pull request by dpeng817**
## Summary
This diff permits dagit / workspaces to target a bare graph. (Fix for https://github.com/dagster-io/dagster/issues/4449).

## Test Plan
Confirmed that targeting a bare graph works in dagit (taken to same page as if you had just a bare graph in a repository).

I'm a bit ignorant about the testing structure of this piece of the codebase, so if anyone has advice for more robust tests, that would be welcome.

## Checklist
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by gibsondan — COMMENTED** [review]
this looks good to me but I don't have full CRAG context

I see a test_autodiscovery.py file that seems like a good place to add a new test

**Comment by sryza** [comment]
+1 to adding to test_autodiscovery.py.  Otherwise, LGTM.

File: python_modules/dagster/dagster/core/definitions/event_metadata.py [hunk 1]
```diff
@@ -9,41 +9,41 @@
...
-        "PathMetadataEntryData",
-        "JsonMetadataEntryData",
-        "MarkdownMetadataEntryData",
-        "FloatMetadataEntryData",
-        "IntMetadataEntryData",
-        "PythonArtifactMetadataEntryData",
-        "DagsterAssetMetadataEntryData",
-        "DagsterPipelineRunMetadataEntryData",
-    ]
+ParseableMetadataEntryData = Union[
```
Comment (sryza) [inline python_modules/dagster/dagster/core/definitions/event_metadata.py:12]: Is this metadata stuff related to auto-discovery of bare graphs?

File: python_modules/dagster/dagster_tests/cli_tests/workspace_tests/autodiscovery_tests/single_graph.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from dagster import graph
+
+
+@graph
+def graph_one():
+    pass
```
Comment (sryza) [inline python_modules/dagster/dagster_tests/cli_tests/workspace_tests/autodiscovery_tests/single_graph.py:6]: Nitpick: include newline at end of file.

File: python_modules/dagster/dagster/core/definitions/event_metadata.py (hunk 1 above)
Comment (dpeng817) [reply python_modules/dagster/dagster/core/definitions/event_metadata.py:12]: woah I don't know what happened here. Must've gotten foisted in by accident.

**Review by sryza — APPROVED** [review]
LGTM once this passes buildkite
